### DISCUSSION TITLE
Cannot use astro.config.ts

### DISCUSSION THREAD

**Discussion by bdphil86** [nxtensions/nxtensions#286] [Q&A] [answered]
On the astro.build website, it says that we can use an astro.config.ts file, but this doesn't appear to work. Event if I use the generators and specify the config, it says that it cannot find the file.

Is there anything special that needs to be done to support this?

**Comment by leosvelperez** [comment, marked as answer]
hey @bdphil86!

Currently, the executors only support the `astro.config.mjs` file automatically. If you use a different one, you must specify the `config` option in your project configuration targets (a path relative to the project root). For example:

```json
{
  ...
  "targets": {
    "build": {
      "executor": "@nxtensions/astro:build",
      "options": {
        "config": "astro.config.ts"
      }
    },
    ...
  }
}
```

That said, I think it would be nice if the default supported configuration filenames were to be handled by default. I added an issue for that https://github.com/nxtensions/nxtensions/issues/287.

**Reply by leosvelperez** [reply]
@bdphil86 a fix for the issue was released on [v3.3.0](https://github.com/nxtensions/nxtensions/releases/tag/%40nxtensions%2Fastro%40v3.3.0). You can now use any of the automatically supported Astro config filenames (`astro.config.js`, `astro.config.mjs`, `astro.config.cjs` and `astro.config.ts`) without specifying the filename in the target options. The target option is only required for filenames different than those automatically supported by Astro.

**Reply by bdphil86** [reply]
Thanks a bunch, @leosvelperez! I'll try that out this afternoon for the project I'm working on!
